File: e-commerce-backend/src/main/java/com/tientrannnn/controllers/AuthController.java
package com.tientrannnn.controllers;

import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;
import com.tientrannnn.dtos.AuthController.LoginRequest;
import com.tientrannnn.dtos.AuthController.SessionResponse;
import com.tientrannnn.models.User;
import com.tientrannnn.repositories.UserRepository;
import jakarta.servlet.http.HttpSession;
import org.mindrot.jbcrypt.BCrypt;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.GetMapping;

@RestController
@RequestMapping("/api/auth")
public class AuthController {

  @Autowired
  private UserRepository UserRepository;

  @GetMapping("info-account")
  public ResponseEntity<?> infoAccount(HttpSession session) {
    User user = (User) session.getAttribute("user");
    if (user == null) {
      return ResponseEntity.status(401).body("Unauthorized");
    }
    return ResponseEntity.ok(new SessionResponse(user));
  }

  @PostMapping("login")
  public ResponseEntity<?> login(@RequestBody LoginRequest request, HttpSession session) {
    String email = request.getEmail();
    String password = request.getPassword();
    User user = UserRepository.findByEmail(email);
    if (user == null) {
      return ResponseEntity.status(401).body("Email does not exist");
    }
    if (!BCrypt.checkpw(password, user.getPassword())) {
      return ResponseEntity.status(401).body("Incorrect password");
    }
    session.setAttribute("user", user);
    return ResponseEntity.ok("Login successful");
  }

  @PostMapping("logout")
  public ResponseEntity<?> logout(HttpSession session) {
    session.removeAttribute("user");
    return ResponseEntity.ok("Logout successful");
  }
}

File: e-commerce-frontend/src/api/AuthController.js
import axios from 'axios'

export async function login(email, password) {
  try {
    await axios.post('/api/auth/login', { email, password })
  } catch (error) {
    console.log('Login failed:', error)
    throw error
  }
}

export async function getInfoAccount() {
  try {
    const response = await axios.get('/api/auth/info-account')
    return response.data
  } catch (error) {
    console.log('Failed to get account info')
    throw error
  }
}

export const logout = async () => {
  try {
    await axios.post('/api/auth/logout')
  } catch (error) {
    console.log('Logout failed:', error)
    throw error
  }
}
